Automattic/pocket-casts-android
# Slumber Studios - Update What's New layout

## Description
This updates `What's New` layout for `Slumber Studios` promo. 

- Currently, it reuses existing What's New popup layout. If there's time, I'll migrate it to a modal sheet
- Button texts and actions to be updated based on eligibility 

## Testing Instructions
1. Apply [this patch](https://github.com/Automattic/pocket-casts-android/files/14176680/patch_whats_new.patch)
2. Install debugProd build
3. Launch the app
4. ✅ Notice what's new for `Slumber Studios` promo is displayed
5. ✅ Notice the sway animation
6. Tap on Learn more
7. ✅ Notice that `Slumber Studios` site is opened

** Since we're testing on a debugProd build variant, promo code will be empty in the What's New body text.

## Screenshots or Screencast 

https://github.com/Automattic/pocket-casts-android/assets/1405144/de3d0611-ba16-4e41-a4ed-9e48ffe2f4f8

From Compose Previews
1| 2| 3| 4
---|---|---|---
<img width="178" alt="Screenshot 2024-02-06 at 1 42 11 PM" src="https://github.com/Automattic/pocket-casts-android/assets/1405144/013e7613-7ef9-4058-aa09-3c2b275c92f2"> | <img width="180" alt="Screenshot 2024-02-06 at 1 42 02 PM" src="https://github.com/Automattic/pocket-casts-android/assets/1405144/32a5e15b-1185-4ed9-bd01-b26ce5313a6a"> | <img width="180" alt="Screenshot 2024-02-06 at 1 41 47 PM" src="https://github.com/Automattic/pocket-casts-android/assets/1405144/16a19fe7-680e-4852-93dc-624710128e47">|<img width="181" alt="Screenshot 2024-02-06 at 1 41 36 PM" src="https://github.com/Automattic/pocket-casts-android/assets/1405144/83c1da86-a62b-416e-b080-0078f473cc60">



## Checklist
 
#### I have tested any UI changes...
